web-common/src/features/dashboards/state-managers/loaders/most-recent-explore-state.ts

```diff
@@ -0,0 +1,97 @@
+import { validateAndCleanExploreState } from "@rilldata/web-common/features/dashboards/stores/validate-and-clean-explore-state";
+import { DashboardState_ActivePage } from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+import type {
+  V1ExploreSpec,
+  V1MetricsViewSpec,
+} from "@rilldata/web-common/runtime-client";
+import type { MetricsExplorerEntity } from "../../stores/metrics-explorer-entity";
+
+function getKeyForLocalStore(
+  exploreName: string,
+  storageNamespacePrefix: string | undefined,
+) {
+  return `rill:app:explore:${storageNamespacePrefix ?? ""}${exploreName}`.toLowerCase();
+}
+
+export function getMostRecentPartialExploreState(
+  exploreName: string,
+  storageNamespacePrefix: string | undefined,
+  metricsViewSpec: V1MetricsViewSpec,
+  exploreSpec: V1ExploreSpec,
+) {
+  const key = getKeyForLocalStore(exploreName, storageNamespacePrefix);
+  try {
+    const rawExploreState = localStorage.getItem(key);
+    if (!rawExploreState) {
+      // Return this so that destructuring is simple
+      return { mostRecentPartialExploreState: undefined, errors: [] };
+    }
+
+    const stateFromLocalStorage = JSON.parse(
+      rawExploreState,
+    ) as Partial<MetricsExplorerEntity>;
+    const errors = validateAndCleanExploreState(
+      metricsViewSpec,
+      exploreSpec,
+      stateFromLocalStorage,
+    );
+    return { mostRecentPartialExploreState: stateFromLocalStorage, errors };
+  } catch {
+    // no-op
+  }
+  // Return this so that destructuring is simple
+  return { mostRecentPartialExploreState: undefined, errors: [] };
+}
+
+export function saveMostRecentExploreState(
+  exploreName: string,
+  storageNamespacePrefix: string | undefined,
+  exploreState: MetricsExplorerEntity,
+) {
```

Comment:
This function name implies that the _full_ `ExploreState` object is saved, not a subset. Can you update the name + add a function-level comment (to explain the "why") to make it more clear that only a subset is saved?